### PR TITLE
Enable nullability in some DataGridView.Methods members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some DataGridView.Methods members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10463)

### CODE REVIEW

**Comment by gpetrou** [comment]
@lonitra any feedback on this one?

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridView.Methods.cs [hunk 1]
```diff
@@ -19314,7 +19334,7 @@ internal void OnSortGlyphDirectionChanged(DataGridViewColumnHeaderCell dataGridV
             if (dataGridViewColumnHeaderCell.SortGlyphDirection == SortOrder.None)
             {
                 SortedColumn = null;
-                DataGridViewColumn dataGridViewColumn = dataGridViewColumnHeaderCell.OwningColumn;
+                DataGridViewColumn dataGridViewColumn = dataGridViewColumnHeaderCell.OwningColumn!;
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridView.Methods.cs:19337]: Can we handle null here by checking in the if statement below?

**Review by lonitra — COMMENTED** [review]
1 small comment otherwise looks good!

**Review by lonitra — APPROVED** [review]
🚀

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridView.Methods.cs [hunk 2]
```diff
@@ -19293,9 +19302,20 @@ protected virtual void OnSelectionChanged(EventArgs e)
...
-        DataGridViewSortCompareEventArgs dgvsce = new DataGridViewSortCompareEventArgs(dataGridViewSortedColumn, value1, value2, rowIndex1, rowIndex2);
+    internal bool OnSortCompare(
+        DataGridViewColumn dataGridViewSortedColumn,
+        object? value1,
+        object? value2,
+        int rowIndex1,
+        int rowIndex2,
+        out int sortResult)
+    {
+        DataGridViewSortCompareEventArgs dgvsce = new DataGridViewSortCompareEventArgs(
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridView.Methods.cs:19313]: In the next PR in this file, please replace this with target-types `new`, I do not want to block this one from merging.

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**